### ISSUE TITLE
Error loading module with assets on Windows

### ISSUE THREAD

**Issue by rmcshane** [systemjs/systemjs#751]
Loading a module with assets on Windows fails.  This is caused by line 2633:
```
if (typeof window != 'undefined' && typeof document != 'undefined' && window.location)
  var windowOrigin = location.protocol + '//' + location.hostname + (location.port ? ':' + location.port : '');
```

In this case, `location.protocol = 'file:'` which causes problems at line 2687:
```
if (windowOrigin && address.substr(0, windowOrigin.length) === windowOrigin) {
  address = address.substr(windowOrigin.length);
  dirname = dirname.substr(windowOrigin.length);
}
```

Which causes `address = '/c:/path/to/module'`.  This eventually causes the module to treat `__dirname = 'c:\c:\path\to\module'` and loading assets fails.

The following change fixes this issue at line 2633:
```
if (typeof window != 'undefined' && typeof document != 'undefined' && window.location && location.hostname)
  var windowOrigin = location.protocol + '//' + location.hostname + (location.port ? ':' + location.port : '');
```

**Comment by guybedford** [comment]
Thanks for sharing. Does the change:

```javascript
if (typeof window != 'undefined' && typeof document != 'undefined' && window.location)
  var windowOrigin = location.protocol + '//' + (location.protocol == 'file' && location.hostname[0] != '/' ? '/' : '') + location.hostname + (location.port ? ':' + location.port : '');
```

also work for you here?

**Comment by rmcshane** [comment]
That nearly fixes it.  The protocol includes a colon, so adding that to the check for a file protocol fixes the issue I noticed:
```
if (typeof window != 'undefined' && typeof document != 'undefined' && window.location)
  var windowOrigin = location.protocol + '//' + (location.protocol == 'file:' && location.hostname[0] != '/' ? '/' : '') + location.hostname + (location.port ? ':' + location.port : '');
```

This problem doesn't happen on Mac, or at least doesn't in my environment.

**Comment by guybedford** [comment]
Excellent, thanks I've included the fix.

**Comment by guybedford** [comment]
The fix wasn't quite right - I've corrected it in https://github.com/systemjs/systemjs/commit/91b7bc01cbfc71872cb69811e583ca71e9036a26. Do let me know if you still have any problems here.

**Comment by guybedford** [comment]
NB this should be ported for 0.18 before the 0.19 release.

**Comment by guybedford** [comment]
Released in 0.19.0.
